### PR TITLE
Webengine misc cleanup.

### PR DESCRIPTION
**Pull request by Teklad**
Just some minor cleanup, removing unused headers... mild tweaks to code to use a few more C++11 constructs.  And a small bug fixes where I forgot to include "real" files in the DocEngine work-around.

On a side note, we probably need to find a better way to do on_proxyMessageReceived... its starting to look pretty ugly.  Might write a "function map", which would route the message to the appropriate function.  This would be similar to how we do it on the JS side.

Another side note: I also came up with a way to delay resizing of the QWebEngineView until the resize is finished (buttery smooth resize)... but its a little hacky so I'm going to try a few different things to see what works best there.  

Mainly getting the ACTUAL state of the mouse after a resize event has completed is darn near impossible using your typical QT functions.  QApplication::mouseButtons() returns the state when the event started, unfortunately.

We could use QAbstractEventDispatcher, but we'd need to write separate bits of code for each individual OS(Windows/Mac/Linux).

Setting up an app-wide eventFilter and keeping track of "is_resizing" would probably be a good way to go, though I'm not 100% on that.

### CODE REVIEW

**Comment by Teklad** [comment]
@danieleds, you alive with us? :)

File: src/ui/EditorNS/editor.cpp [hunk 1]
```diff
@@ -656,8 +646,8 @@ namespace EditorNS
             return theme;
 
         auto editorPath = QFileInfo(Notepadqq::editorPath());
-        auto bundledThemesDir = QDir(editorPath.absolutePath() + "/libs/codemirror/theme/");
-        auto themeFile = bundledThemesDir.filePath(name + ".css");
+        auto bundledThemesDir = QDir(editorPath.absolutePath() % "/libs/codemirror/theme/");
+        auto themeFile = bundledThemesDir.filePath(name % ".css");
```
Comment (danieleds) [inline src/ui/EditorNS/editor.cpp]: Same as before. Is this giving a visible performance improvement?

File: src/ui/main.cpp [hunk 2]
```diff
@@ -101,7 +101,7 @@ int main(int argc, char *argv[])
 
     // There are no other instances: start a new server.
     a.startServer();
-    Editor::addEditorToBuffer();
+    //Editor::addEditorToBuffer();
```
Comment (danieleds) [inline src/ui/main.cpp]: mistake? Or was it making performance worse?

**Comment by danieleds** [comment]
Still alive, sorry for the long wait!

File: src/ui/main.cpp (hunk 2 above)
Comment (Teklad) [inline src/ui/main.cpp]: It no longer has any effect on performance and just eats more memory/increases startup time.

File: src/ui/EditorNS/editor.cpp (hunk 1 above)
Comment (JuBan1) [inline src/ui/EditorNS/editor.cpp]: I don't think using `%` will give any improvements here. `QStringBuilder` delays construction of the `QString` until all concatenations a processed. But with just one concatenation there's nothing to delay.

Comment (Teklad) [inline src/ui/EditorNS/editor.cpp]: You're more than likely right... not sure what I was thinking here.

**Comment by Teklad** [comment]
Alright... reverted the QStringBuilder change since it probably doesn't have any performance improvement.  The last bit was more-so to test and see if it made any difference related to startup/runtime performance originally.

The consensus I got on it was it no longer gives any performance increase at all... and actually causes notepadqq to eat more memory, which isn't desirable at this point.

I'm not really sure what else I can do for QWebEngine at this point outside of minor pokes and prods as they are requested.  It's (mostly) feature complete... and any remaining bugs are those that probably shouldn't be fixed in this scope. (DocEngine and Editor overhauls are way past due)

**Comment by danieleds** [comment]
Perfect!

**Pull request merged**